src/databricks/labs/blueprint/paths.py

```diff
@@ -448,17 +441,296 @@ def match(self, path_pattern, *, case_sensitive=None):
+        # The cached _file_info value for the instance.
+        "_cached_file_info",
+    )
+    _cached_file_info: FileInfo
+
+    @classmethod
+    def _from_file_info(cls, ws: WorkspaceClient, file_info: FileInfo) -> DBFSPath:
+        """Special (internal-only) constructor that creates an instance based on FileInfo."""
+        if not file_info.path:
+            msg = f"Cannot initialise without file path: {file_info}"
+            raise ValueError(msg)
+        path = cls(ws, file_info.path)
+        path._cached_file_info = file_info
+        return path
+
+    as_uri = _na("as_uri")
+
+    def as_fuse(self) -> Path:
         """Return FUSE-mounted path in Databricks Runtime."""
         if "DATABRICKS_RUNTIME_VERSION" not in os.environ:
             logger.warning("This method is only available in Databricks Runtime")
-        return Path("/Workspace", self.as_posix().lstrip("/"))
+        return Path("/dbfs", self.as_posix().lstrip("/"))
 
-    def home(self):  # pylint: disable=arguments-differ
-        """Return the user's home directory. Adapted from pathlib.Path"""
-        return WorkspacePath(self._ws, "~").expanduser()
+    def exists(self, *, follow_symlinks: bool = True) -> bool:
+        """Return True if the path points to an existing file, directory, or notebook"""
+        if not follow_symlinks:
+            raise NotImplementedError("follow_symlinks=False is not supported for DBFS")
+        try:
+            self._ws.dbfs.get_status(self.as_posix())
+            return True
+        except NotFound:
+            return False
+
+    def mkdir(self, mode: int = 0o600, parents: bool = True, exist_ok: bool = True) -> None:
+        """Create a directory in DBFS. Only mode 0o600 is supported."""
+        if not exist_ok:
+            raise ValueError("exist_ok must be True for Databricks Workspace")
+        if not parents:
+            raise ValueError("parents must be True for Databricks Workspace")
+        if mode != 0o600:
+            raise ValueError("other modes than 0o600 are not yet supported")
+        self._ws.dbfs.mkdirs(self.as_posix())
+
+    def rmdir(self, recursive: bool = False) -> None:
+        """Remove a DBFS directory"""
+        self._ws.dbfs.delete(self.as_posix(), recursive=recursive)
+
+    def _rename(self: P, target: str | bytes | os.PathLike, overwrite: bool) -> P:
+        dst = self.with_segments(target)
+        if overwrite:
+            with dst.open(mode="wb") as writer, self.open(mode="rb") as reader:
+                shutil.copyfileobj(reader, writer, length=1024 * 1024)
+            self.unlink()
+        else:
+            self._ws.dbfs.move(self.as_posix(), dst.as_posix())
+        return dst
+
+    def unlink(self, missing_ok: bool = False) -> None:
+        """Remove a file in Databricks Workspace."""
+        if not missing_ok and not self.exists():
+            raise FileNotFoundError(f"{self.as_posix()} does not exist")
+        self._ws.dbfs.delete(self.as_posix())
```

Comment:
```suggestion
        try:
          self._ws.dbfs.delete(self.as_posix())
        except NotFound as e:
          if not missing_ok: 
            raise e
```
save API calls

Reply:
It turns out that the DBFS API client doesn't report an error if the target doesn't exist, so I've left this in place with a comment explaining why.

I've updated the `WorkspacePath` version to follow this pattern though: errors are reported if the target doesn't exist so we can handle it in the unhappy path. Aside from the extra calls, it avoids an avoidable race-condition.

Integration tests now cover this behaviour.